File: 0225-implement-stack-using-queues/0225-implement-stack-using-queues.py
class MyStack:

    def __init__(self):
        self.q = deque()

    def push(self, x: int) -> None:
        self.q.append(x)

    def pop(self) -> int:
        l = len(self.q)
        while l >= 2:
            self.q.append(self.q.popleft())
            l -= 1
        return self.q.popleft()

    def top(self) -> int:
        val = self.pop()
        self.push(val)
        return val

    def empty(self) -> bool:
        return len(self.q) == 0


# Your MyStack object will be instantiated and called as such:
# obj = MyStack()
# obj.push(x)
# param_2 = obj.pop()
# param_3 = obj.top()
# param_4 = obj.empty()

File: 0225-implement-stack-using-queues/0225-implement-stack-using-queues.go
type MyStack struct {
    queue []int
}


func Constructor() MyStack {
    return MyStack{
        queue:make([]int, 0),
    }
}


func (this *MyStack) Push(x int)  {
    this.queue=append(this.queue, x)
}


func (this *MyStack) Pop() int {
    n:=len(this.queue)-1
    for n!=0{
        val:=this.queue[0]
        this.queue=this.queue[1:]
        this.queue=append(this.queue, val)
        n--
    }
    val:=this.queue[0]
    this.queue=this.queue[1:]
    return val
}


func (this *MyStack) Top() int {
    val:=this.Pop()
    this.Push(val)
    return val
}


func (this *MyStack) Empty() bool {
    return len(this.queue)==0
}


/**
 * Your MyStack object will be instantiated and called as such:
 * obj := Constructor();
 * obj.Push(x);
 * param_2 := obj.Pop();
 * param_3 := obj.Top();
 * param_4 := obj.Empty();
 */